examples/operator.rs

```diff
@@ -0,0 +1,100 @@
+extern crate timely;
+extern crate timely_communication;
+
+use std::collections::HashMap;
+
+use timely_communication::Configuration;
+use timely::dataflow::operators::{ToStream, Input, Inspect, Operator, FrontierNotificator};
+use timely::dataflow::channels::pact::Pipeline;
+use timely::progress::timestamp::RootTimestamp;
+use timely::dataflow::Scope;
+
+fn main() {
+    timely::execute(Configuration::Thread, |root| {
+        root.scoped(|scope| {
+            (0u64..10).to_stream(scope)
+                .unary_frontier(Pipeline, "example", |mut builder| {
+                    let mut cap = Some(builder.get_cap(RootTimestamp::new(12)));
+                    move |(input, _frontier), output| {
+                        cap = None;
+                        while let Some((time, data)) = input.next() {
+                            output.session(&time).give_content(data);
+                        }
+                    }
+                }).inspect(|x| println!("{:?}", x));
+        });
+    }).unwrap();
+
+    timely::execute(Configuration::Thread, |root| {
+        root.scoped(|scope| {
+            (0u64..10).to_stream(scope)
+                .unary(Pipeline, "example", |mut builder| {
```

Comment:
If you're still interested in something like this, I think I have a new plan: instead of getting a builder (which complicates things), you request a capability (with its timestamp) and you get the actual cap in the closure. Something like:

```
.unary(Pipeline, "example", Some(RootTimestamp::new(12)), |mut cap| {
  // cap is now a Option<Capability> = Some(Capability for timestamp 12)
  // you then `move` it into the actual operator logic

  move |input, output| {
    if let Some(ref c) = cap.take() {
      // ...
```

The outer closure can still be used to instantiate a notificator.

```
         in1.unary_frontier(&in2, Pipeline, Pipeline, "example", None, |_| {
            let mut notificator = FrontierNotificator::new();
            move |(input1, frontier1), output| {
              for time in notificator.iter(&[frontier1]) {
```

This could (probably) be improved by handing the reference to the frontier/progress updates to notificator directly, instead of having to refer to it from inside the operator logic.

```
         in1.unary_frontier(&in2, Pipeline, Pipeline, "example", None, |_, frontier1| {
            let mut notificator = FrontierNotificator::new(frontier1);
            move |input1, output| {
              for time in notificator.iter() {
```

If this sounds reasonable, let me know and I can take a stab at it. But if doesn't line up with what you have in mind, feel free to close this PR.